sys/pm_callouts.c

```diff
@@ -813,30 +720,24 @@ void classifyAll(portmaster_packet_info* packetInfo,
         // DEBUG: print its TCP 4-tuple
         INFO("Getting verdict for %s", print_packet_info(packetInfo));
 
-        //Block, Absorb, and Copy Packet to packet_cashe
-        //according to https://docs.microsoft.com/en-us/windows-hardware/drivers/network/types-of-callouts
-        classifyOut->actionType = FWP_ACTION_BLOCK;
-        classifyOut->flags|= FWPS_CLASSIFY_OUT_FLAG_ABSORB;   //Set Absorb Bit 1
-        classifyOut->rights&= ~FWPS_RIGHT_ACTION_WRITE;       //Set Write Bit 0
-
         //Inbound traffic requires special treatment - this bitshifterei is a special source of error ;-)
         if (packetInfo->direction == 1) { //Inbound
             status = NdisRetreatNetBufferDataStart(nb, ipHeaderSize, 0, NULL);
             if (!NT_SUCCESS(status)) {
                 ERR("Argh!!!!");
-                return;
+                return FWP_ACTION_NONE;
             }
         }
 
         status = copy_packet_data_from_nb(nb, 0, &data, &data_len);
         if (!NT_SUCCESS(status)) {
             ERR("AAAA!!! copy_packet_data_from_nb 2: %d", status);
-            return;
+            return FWP_ACTION_NONE;
         }
 
         INFO("copy_packet_data_from_nb rc=%d, data_len=%d", status, data_len);
 
-        //Even though nb is not used anymore (?) we still shift back (because we have a problem, which we do not understand)
+        // In order to be as clean as possible, we shift back nb, even though it may not be necessary.
         if (packetInfo->direction == 1) { //Inbound
```

Comment:
Consider using a macro for that like `IS_INBOUND(packetInfo->direction)`. It's easier to read and you don't need to repeast the `// Inbound` comments